src/main/java/org/openhab/tools/analysis/checkstyle/ImportExportedPackagesCheck.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.ivy.osgi.core.ManifestParser;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+/**
+ * Checks if all of the exported packages are imported by the bundle itself
+ *
+ * @author Mihaela Memova
+ */
+public class ImportExportedPackagesCheck extends AbstractStaticCheck {
+
+	public static final String NOT_IMPORTED_PACKAGE_MESSAGE = "The exported package is not imported";
+	public static final String MANIFEST_EXTENSTION = "MF";
+	public static final String EXPORT_PACKAGES_HEADER = "Export-Package:";
+
+	public ImportExportedPackagesCheck() {
+		setFileExtensions(MANIFEST_EXTENSTION);
+	}
+
+	@Override
+	protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+		int exportsStartIndex = 0;
+		for (int i = 0; i < lines.size(); i++) {
+			if (lines.get(i).startsWith(EXPORT_PACKAGES_HEADER)) {
+				exportsStartIndex = i;
+			}
+		}
+
+		try {
+			Set exports = ManifestParser.parseManifest(file).getExports();
+			Set imports = ManifestParser.parseManifest(file).getImports();
+
+			for (Object export : exports) {
+				if (!setContainsObject(imports, export)) {
+					int lineToLog = findLineNumber(lines, export.toString(), exportsStartIndex);
+					if(lineToLog != -1) {
+						log(lineToLog, NOT_IMPORTED_PACKAGE_MESSAGE);
+					}
+				}
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+			log(0, String.format("An exception %s was thrown while processing the file: ", e, file.getPath()));
+		} catch (ParseException e) {
+			log(0, String.format("An exception %s was thrown while trying to parse the MANIFEST: ", e, file.getPath()));
+		}
+	}
+
+	private boolean setContainsObject(Set set, Object obj) {
+		for (Object o : set) {
+			if ((o.toString()).equals(obj.toString())) {
```

Comment:
_Maybe you can add a small comment what you are actually doing here, I read the underlying source code and it makes complete sense what you do here, but the code first made me curious what was going on and why the normal Set.contains was not used :-)_